Tests/Realm.Tests/SourceGeneration/TestClasses/ManuallyGeneratedClass.cs

```diff
@@ -0,0 +1,270 @@
+        {
+            var unmanagedAccessor = _accessor;
+            _accessor = (IManuallyGeneratedClassAccessor)managedAccessor;
+
+            if (helper != null)
+            {
+                IntValue = unmanagedAccessor.IntValue;
+                StringValue = unmanagedAccessor.StringValue;
+
+                if (!skipDefaults)
+                {
+                    ListValue.Clear();
+                }
+
+                foreach (var val in unmanagedAccessor.ListValue)
+                {
+                    ListValue.Add(val);
+                }
+            }
+
+            if (_propertyChanged != null)
+            {
+                SubscribeForNotifications();
+            }
+
+            OnManaged();
+        }
+        #endregion
+
+        private event PropertyChangedEventHandler _propertyChanged;
+
+        public event PropertyChangedEventHandler PropertyChanged
+        {
+            add
+            {
+                if (IsManaged && _propertyChanged == null)
+                {
+                    SubscribeForNotifications();
+                }
+
+                _propertyChanged += value;
+            }
+
+            remove
+            {
+                _propertyChanged -= value;
+
+                if (IsManaged &&
+                    _propertyChanged == null)
+                {
+                    UnsubscribeFromNotifications();
+                }
+            }
+        }
+
+        protected virtual void OnPropertyChanged(string propertyName)
+        {
+        }
+
+        protected void RaisePropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            _propertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+            OnPropertyChanged(propertyName);
+        }
+
+        protected internal virtual void OnManaged()
+        {
+        }
+
+        private void SubscribeForNotifications()
+        {
+            _accessor.SubscribeForNotifications(RaisePropertyChanged);
+        }
+
+        private void UnsubscribeFromNotifications()
+        {
+            _accessor.UnsubscribeFromNotifications();
+        }
+    }
+}
+
+// Having a separate namespace allows to hide the implementation details better.
+namespace Realm.Generated
+{
+    internal class ManuallyGeneratedClassObjectHelper : IRealmObjectHelper
+    {
+        public void CopyToRealm(IRealmObjectBase instance, bool update, bool skipDefaults)
+        {
+            throw new InvalidOperationException("This method should not be called for source generated classes.");
+        }
+
+        public ManagedAccessor CreateAccessor() => new ManuallyGeneratedClassManagedAccessor();
+
+        public IRealmObjectBase CreateInstance()
+        {
+            return new ManualllyGeneratedClass();
+        }
+
+        public bool TryGetPrimaryKeyValue(IRealmObjectBase instance, out object value)
+        {
+            value = (instance.Accessor as IManuallyGeneratedClassAccessor).PrimaryKeyValue;
```

Comment:
I don't like using `as` casts like that. Can we do a regular one so that we throw properly?